### PR TITLE
ec_deployment: Remove APM secret_token from config

### LINKED ISSUE #56: ec_deployment: Add apm_secret_token as top level field
## Overview

Currently the APM `secret_token` is stored under `ec_deployment<name>.apm.0.config.secret_token`.
While this is fine, the ability to obtain `secret_token` through a `GET /deployment/<id>` might
change in the future, thus is better to parse the output of the create / update deployment and store
the value in a top level field named `apm_secret_token`.

### PR DESCRIPTION
**Pull request by marclop**
## Description

Removes the `secret_token` and `secret_token` parsing from the `config`
blob under the `apm` object.

Instead of relying on the `GET /deployment/<id>` to populate the field,
it is instead parsed from the Create / Update API response, saving
it to a key named `apm_secret_token`.

Updates the tests and the defined Deployment version.

## Related Issues

Resolves #56 

## Types of Changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by marclop** [comment]
```console
$ make testacc
-> Formatting Go files...
-> Formatting Terraform documentation blocks...
-> Done.
-> Running terraform acceptance tests...
=== RUN   TestAccDatasourceDeployment_basic
=== PAUSE TestAccDatasourceDeployment_basic
=== RUN   TestAccDeployment_basic
=== PAUSE TestAccDeployment_basic
=== RUN   TestAccDeploymentTrafficFilter_basic
=== PAUSE TestAccDeploymentTrafficFilter_basic
=== CONT  TestAccDatasourceDeployment_basic
=== CONT  TestAccDeploymentTrafficFilter_basic
=== CONT  TestAccDeployment_basic
--- PASS: TestAccDeploymentTrafficFilter_basic (13.89s)
--- PASS: TestAccDatasourceDeployment_basic (575.68s)
--- PASS: TestAccDeployment_basic (954.22s)
PASS
ok  	github.com/terraform-providers/terraform-provider-ec/ec/acc	955.125s
```

File: ec/ecresource/deploymentresource/schema.go [hunk 1]
```diff
@@ -56,6 +56,13 @@ func NewSchema() map[string]*schema.Schema {
 			Sensitive: true,
 		},
 
+		// APM secret_token
+		"apm_secret_token": {
```
Comment (karencfv) [inline ec/ecresource/deploymentresource/schema.go:60]: Not against this change but wondering why are we moving it up one level?

Comment (marclop) [reply ec/ecresource/deploymentresource/schema.go:60]: it's to make it like the `elasticsearch_username` and `elasticsearch_password ` fields (same file L49-55)

**Review by karencfv — APPROVED** [review]
Thanks for answering my question!

**Pull request merged**